src/tiffimage_int.cpp

```diff
@@ -1696,6 +1696,7 @@ namespace Exiv2 {
             pSourceDir->accept(finder);
             TiffEntryBase* te = dynamic_cast<TiffEntryBase*>(finder.result());
             if (   te
+                && te->pValue()
```

Comment:
I'd prefer to have `te->pValue()` as temporary here.

```cpp
const auto& pV = te != nullptr ? te->pValue() : nullptr;
if (pV && pV->typeID() && ...)
```

Reply:
I'll do it 😉

Reply:
done